### PR TITLE
attach: set the terminal size and handle SIGWINCH

### PR DESCRIPTION
**Pull request by giuseppe**
Notify conmon when the terminal size changes.  Use the same notification
to set the correct initial size.

Closes: https://github.com/projectatomic/libpod/issues/351

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: libpod/container_attach.go [hunk 1]
```diff
@@ -64,8 +64,15 @@ func (c *Container) attachContainerSocket(resize <-chan remotecommand.TerminalSi
 	defer controlFile.Close()
 
 	kubecontainer.HandleResizing(resize, func(size remotecommand.TerminalSize) {
+		controlPath := filepath.Join(c.bundlePath(), "ctl")
+		controlFile, err := os.OpenFile(controlPath, unix.O_WRONLY, 0)
```
Comment (mheon) [inline libpod/container_attach.go]: Don't like the naming here - you're shadowing `controlFile` in the function calling `HandleResizing`. `ctrControlFile` might be better?

Comment (giuseppe) [reply libpod/container_attach.go]: I'll drop the `controlFile` block in the function as it is not used anymore.

File: libpod/container_attach.go [hunk 2]
```diff
@@ -56,16 +56,16 @@ func (c *Container) attachContainerSocket(resize <-chan remotecommand.TerminalSi
...
-	if err != nil {
-		return errors.Wrapf(err, "failed to open container ctl file")
-	}
-	defer controlFile.Close()
-
 	kubecontainer.HandleResizing(resize, func(size remotecommand.TerminalSize) {
+		controlPath := filepath.Join(c.bundlePath(), "ctl")
+		controlFile, err := os.OpenFile(controlPath, unix.O_WRONLY, 0)
+		if err != nil {
+			return
```
Comment (rhatdan) [inline libpod/container_attach.go]: Should we logrus.Warn Here?

File: libpod/container_attach.go [hunk 3]
```diff
@@ -56,16 +56,16 @@ func (c *Container) attachContainerSocket(resize <-chan remotecommand.TerminalSi
...
+		}
+		defer controlFile.Close()
+
 		logrus.Debugf("Received a resize event: %+v", size)
-		_, err := fmt.Fprintf(controlFile, "%d %d %d\n", 1, size.Height, size.Width)
+		_, err = fmt.Fprintf(controlFile, "%d %d %d\n", 1, size.Height, size.Width)
```
Comment (rhatdan) [inline libpod/container_attach.go]: Nit: Why not just do 
if _, err := fmt ... ; err !=nil

File: libpod/container_api.go [hunk 4]
```diff
@@ -373,8 +401,11 @@ func (c *Container) Attach(noStdin bool, keys string, attached chan<- bool) erro
 	}
 
 	resize := make(chan remotecommand.TerminalSize)
-	defer close(resize)
-
+	if terminal.IsTerminal(int(os.Stdin.Fd())) {
+		resizeTty(resize)
```
Comment (rhatdan) [inline libpod/container_api.go]: Does this mean we never close the resize?

Comment (giuseppe) [reply libpod/container_api.go]: the ownership passes to `resizeTty` that will close the channel

File: libpod/container_attach.go (hunk 3 above)
Comment (giuseppe) [reply libpod/container_attach.go]: sure, pushed a new version to fix this.

**Comment by rhatdan** [comment]
@cgwalters Could you check if this fixes your issue?

**Comment by cgwalters** [comment]
This seems to work well for me!  Thanks a lot @giuseppe !

**Comment by baude** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**